nlp_primitives/number_of_hashtags.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double, NaturalLanguage
+
+from .count_string import CountString
+
+
+class NumberOfHashtags(CountString):
+    """Determines the number of hashtags in a string.
+
+    Description:
+        Given list of strings, determine the number of hashtags
+        in each string. A hashtag is defined as any word starting
+        with a '#' sign followed by a sequence of alphanumeric characters
+        or underscores.
+
+        If a string is missing, return `NaN`.
+
+    Examples:
+        >>> x = ['#regular#expression', 'this is a string', '###__regular#1and_0#expression']
+        >>> number_of_hashtags = NumberOfHashtags()
+        >>> number_of_hashtags(x).tolist()
+        [2.0, 0.0, 3.0]
+    """
+
+    name = "number_of_hashtags"
+    input_types = [ColumnSchema(logical_type=NaturalLanguage)]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
```

Comment:
I believe we should use `IntegerNullable` here instead of `Double` since we can never have fractional count values, but we could have missing `nan` values.